suspendmanager: take in account wall plans on the tile below (#4468)
DFHack/dfhack: docs/changelog.txt

## Misc Improvements

> - `suspendmanager`: Take in account walls planned on the z-layer below to determine accessibility to a job

Author: Not sure if it is a bug fix or improvement, and the wording is hard here :)

Reviewer: Clearly an improvement, there never was the expectation to handle all cases with our heuristics.

> - `suspendmanager`: Take in account walls planned on the z-layer below to determine accessibility to a job

Reviewer 2: ```suggestion
- `suspendmanager`: Account for walls planned on the z-layer below when determining accessibility to a job
```
